### PR TITLE
show only capability action related checkbox

### PR DESCRIPTION
**Pull request by aidynoJ**
### Demostrate only capability action related checkbox
This PR is in addition to the previous one. 

**Before**
Screenshot 2023-12-13 at 15 42 25

**After**
Screenshot 2023-12-13 at 15 51 11

And also there are readonly view in role details section , 
Screenshot 2023-12-20 at 21 09 24

There are 3 types of capabilities - data, settings, procedural. Each of them has their own action. For data - view, edit, create, delete,manage. For settings - view, edit, manage. For procedural - only 'execute'.  
For the readonly view in roles details and for EditCreateRole, the same component with props(capabilities, readOnly, onChangeCapabilityCheckbox, isCapabilitySelected) defines how we need to handle render our component. Should checkboxes have to be clickable or not, what kind of checkboxes should be visible, for example when we update, create role, only certain capabilities checkboxes should be visible, depending on what the action is. (see screenshot 2)

### CODE REVIEW

File: src/settings/components/Capabilities/CapabilitiesProcedural.js [hunk 1]
```diff
@@ -24,13 +24,16 @@ const CapabilitiesProcedural = ({ content, readOnly, onChangeCapabilityCheckbox
...
     resource:item => item.resource,
-    execute: item => <CheckBoxWithAsterisk
-      aria-describedby="asterisk-policy-desc"
-      aria-label={getCheckBoxAriaLabel('execute', item.resource)}
-      onChange={event => onChangeCapabilityCheckbox(event, item.id, 'execute')}
-      readOnly={readOnly}
-      checked={item.actions.execute}
-    />,
+    execute: item => {
+      if (!readOnly && item.action !== 'execute') return null;
```
Comment (zburke) [inline src/settings/components/Capabilities/CapabilitiesProcedural.js:28]: Can you add a comment explaining why we don't show a checkbox given these conditions? It is not immediately obvious to me how those conditions relate to hiding it.

File: src/settings/components/Capabilities/CapabilitiesSection.js [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
+import { noop } from 'lodash';
 
 import { FormattedMessage } from 'react-intl';
 
 import { CapabilitiesSettings } from './CapabilitiesSettings';
 import { CapabilitiesProcedural } from './CapabilitiesProcedural';
 import { CapabilitiesDataType } from './CapabilitiesDataType';
 
-const CapabilitiesSection = ({ capabilities, readOnly, onChangeCapabilityCheckbox }) => (
+const CapabilitiesSection = ({ capabilities, readOnly, onChangeCapabilityCheckbox, defineIsCapabilitySelected = noop }) => (
```
Comment (zburke) [inline src/settings/components/Capabilities/CapabilitiesSection.js:11]: Other components do not include a default value for this prop. What is special about this one?

File: src/settings/components/Capabilities/CapabilitiesSettings.js [hunk 3]
```diff
@@ -24,30 +24,23 @@ const CapabilitiesSettings = ({ content, readOnly, onChangeCapabilityCheckbox })
     // policies: formatMessage(columnTranslations.policies
   };
 
+  const renderItemActionCheckBox = (item, action) => {
+    if (!readOnly && item.action !== action) return null;
```
Comment (zburke) [inline src/settings/components/Capabilities/CapabilitiesSettings.js:28]: As above: Please explain; it is not immediately obvious to me how the conditions relate to hiding the checkbox.

File: src/hooks/useCapabilities.js [hunk 4]
```diff
@@ -24,7 +24,7 @@ const useCapabilities = () => {
 
   const { data } = useQuery(
     [nameSpace],
-    () => ky.get(`capabilities?limit=${stripes.config.maxUnpagedResourceCount}&query=cql.allRecords=1 sortby applicationId`).json(),
+    () => ky.get(`capabilities?limit=${stripes.config.maxUnpagedResourceCount}&query=cql.allRecords=1 sortby resource`).json(),
```
[the remark this answers is not in the document]
Comment (aidynoJ) [reply src/hooks/useCapabilities.js:27]: Sorry. It's wrong, we need to group by application name first and then sort by resource. I'll handle it

File: src/settings/components/CreateEditRole/CreateEditRole.js [hunk 5]
```diff
@@ -34,9 +34,23 @@ const CreateEditRole = ({ refetch, selectedRole }) => {
...
+    setSelectedCapabilitiesMap({ ...selectedCapabilitiesMap, [id]: event.target.checked });
+  };
+
+  const groupedCapabilitiesByType = useMemo(() => {
+    return getKeyBasedArrayGroup(data, 'type');
+  }, [data]);
 
   const defineCrudOperationTitleId = () => (selectedRole ? 'ui-authorization-roles.crud.editRole' : 'ui-authorization-roles.crud.createRole');
 
+  const defineIsCapabilitySelected = (id) => selectedCapabilitiesMap[id];
```
[the remark this answers is not in the document]
Comment (aidynoJ) [reply src/settings/components/CreateEditRole/CreateEditRole.js]: isCapabilitySelected is okay, thank you

File: src/settings/components/Capabilities/CapabilitiesSection.js (hunk 2 above)
Comment (aidynoJ) [reply src/settings/components/Capabilities/CapabilitiesSection.js]: Fixed, it was temporary solution

File: src/settings/components/Capabilities/CapabilitiesSettings.js (hunk 3 above)
Comment (aidynoJ) [reply src/settings/components/Capabilities/CapabilitiesSettings.js:39]: Added docstrings

File: src/settings/components/Capabilities/CapabilitiesProcedural.js (hunk 1 above)
Comment (aidynoJ) [reply src/settings/components/Capabilities/CapabilitiesProcedural.js:28]: Added docstrings

**Review by zburke — CHANGES_REQUESTED** [review]
There is one more `CheckBox/Checkbox` change to make and that's the last substantive code change before we can merge. Nice work!

The JSDoc comments are helpful (thank you!) but I still struggle to understand what is happening at a high level. The before/after screenshots in the description show _what_ changes are happening, but there's not a great explanation of _why_ those changes are happening. Without a reference to a ticket where there are more details, or a pointer to documentation that explains how roles and capabilities work, it's just really hard to construct a mental model of what is happening.

If we can include some additional details in the PR description, and make sure that comment is included when the PR is squashed and merged, it will be much appreciated.

File: src/settings/components/Capabilities/CapabilitiesDataType.test.js [hunk 6]
```diff
@@ -1,9 +1,12 @@
 import React from 'react';
+import { MemoryRouter } from 'react-router';
+import userEvent from '@testing-library/user-event';
 
 import {
-  translationsProperties,
   renderWithIntl,
 } from '@folio/stripes-erm-testing';
+
+import { translationsProperties } from '../../../../test/helpers';
```
[the remark this answers is not in the document]
Comment (aidynoJ) [reply src/settings/components/Capabilities/CapabilitiesDataType.test.js:9]: Ok, got it. Thank you, Zak

File: src/settings/components/CreateEditRole/CreateEditRole.js [hunk 7]
```diff
@@ -33,10 +34,29 @@ const CreateEditRole = ({ refetch, selectedRole }) => {
...
+  const { capabilitiesList } = useCapabilities();
+  const { initialRoleCapabilitiesSelectedMap, isSuccess: isRoleCapabilitiesFetched } = useRoleCapabilities(selectedRole?.id);
+  const [selectedCapabilitiesMap, setSelectedCapabilitiesMap] = useState({});
+
+  useEffect(() => {
+    if (isRoleCapabilitiesFetched && selectedRole) {
+      setSelectedCapabilitiesMap(initialRoleCapabilitiesSelectedMap);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isRoleCapabilitiesFetched, selectedRole]);
```
[the remark this answers is not in the document]
Comment (aidynoJ) [reply src/settings/components/CreateEditRole/CreateEditRole.js:53]: Added comments about that

**Pull request closed without merging**